scramjetorg/transform-hub
# STH SD tests

Asana task: https://app.asana.com/0/1200969053491921/1200969053491983

BDD tests for endpoint `/topic/:name` using api-client

to verify:
`NO_HOST=true yarn test:bdd --name="E2E-013"`

with logs:
`NO_HOST=true DEVELOPMENT=true SCRAMJET_TEST_LOG=true yarn test:bdd --name="E2E-013"`


result:
```
Feature: E2E test, where we send and receive data from /topic/:name endpoint by using api-client # features/e2e/E2E-013-topic.feature:1

  @ci @starts-host
  Scenario: E2E-013 TC-001 Send and get data from API STH # features/e2e/E2E-013-topic.feature:4
    Given start host
Starting host on port: 37205
    Then send data "{ \"city\": \"New York\" }" named "cities"
    And get data named "cities"
    Then confirm data defined as "nyc-city" received
Received data:  { "city": "New York" }

  @ci @starts-host
  Scenario: E2E-013 TC-002 Send data via instance, get this data via API # features/e2e/E2E-013-topic.feature:11
    Given start host
Starting host on port: 39783
    And sequence "../packages/reference-apps/endless-names-output.tar.gz" loaded
Package successfully loaded, sequence started.
    And instance started with arguments "10"
    And get data named "names"
    Then confirm data defined as "endless-names-10" received
Received data:  { "name": "Aga" }
{ "name": "Michał" }
{ "name": "Patryk" }
{ "name": "Rafał" }
{ "name": "Aida" }
{ "name": "Basia" }
{ "name": "Natalia" }
{ "name": "Monika" }
{ "name": "Wojtek" }


  @ci @starts-host
  Scenario: E2E-013 TC-003 Send data via API, get this data from the instance # features/e2e/E2E-013-topic.feature:19
    Given start host
Starting host on port: 40935
    And sequence "../packages/reference-apps/hello-input-out.tar.gz" loaded
Package successfully loaded, sequence started.
    And instance started
    Then send data "{ \"name\": \"Hulk\" }" named "names"
    And wait for "1000" ms
    And get output
    Then confirm data defined as "hulkName" received
Received data:  Name is: Hulk


  @ci @starts-host
  Scenario: E2E-013 TC-004 Send data from instance to another instance on the same host # features/e2e/E2E-013-topic.feature:29
    Given start host
Starting host on port: 39971
    And sequence "../packages/reference-apps/endless-names-output.tar.gz" loaded
Package successfully loaded, sequence started.
    And instance started with arguments "10"
    And sequence "../packages/reference-apps/hello-input-out.tar.gz" loaded
Package successfully loaded, sequence started.
    And instance started
    And get output
    Then confirm data defined as "hello-input-out-10" received
Received data:  Name is: Aga
Name is: Michał
Name is: Patryk
Name is: Rafał
Name is: Aida
Name is: Basia
Name is: Natalia
Name is: Monika
Name is: Wojtek


4 scenarios (4 passed)
23 steps (23 passed)
0m56.798s (executing steps: 0m56.748s)
Done in 63.23s.
```
